rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -215,6 +215,13 @@ void Player::play()
   is_in_play_ = true;
   try {
     do {
+      float delay =
```

Comment:
Can you move `delay` out of the loop? I don't think it can change between iterations.

I'd also suggest raising a warning or exception when delay is an invalid value. That way the user knows when they enter an invalid value.

Reply:
I moved it out of the loop. And raising warning and set to 0 if dealy value is invalid. https://github.com/ros2/rosbag2/pull/789/commits/0b4687f53c8c3dde2fa69e5235137a66ff390873